File: zzt-dialogutils/src/main/java/com/zzt/dialogutilcode/MainActivity.java
package com.zzt.dialogutilcode;

import android.content.DialogInterface;
import android.os.Bundle;
import android.util.Log;
import android.view.View;
import android.widget.Toast;

import androidx.activity.result.contract.ActivityResultContracts;
import androidx.appcompat.app.AppCompatActivity;
import androidx.core.content.ContextCompat;
import androidx.recyclerview.widget.RecyclerView;

import com.zzt.adapter.StartActivityRecyclerAdapter;
import com.zzt.dialogutilcode.widget.dialog.QMUIDialog;
import com.zzt.dialogutilcode.widget.dialog.QMUIDialogAction;
import com.zzt.entity.StartActivityDao;

import java.util.ArrayList;
import java.util.List;

public class MainActivity extends AppCompatActivity {
    private static final String TAG = MainActivity.class.getSimpleName();
    RecyclerView rv_list;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        initView();
    }

    private void initView() {
        LeftFixedEditText lfet_text = findViewById(R.id.lfet_text);
        lfet_text.setFixedText("$");

        rv_list = findViewById(R.id.rv_list);

        List<StartActivityDao> mListDialog = new ArrayList<>();
        mListDialog.add(new StartActivityDao("列表对话框", "", "1"));
        mListDialog.add(new StartActivityDao("确定取消对话框", "", "2"));

        StartActivityRecyclerAdapter.setAdapterData(rv_list, RecyclerView.VERTICAL, mListDialog, (itemView, position, data) -> {
            switch (data.getArouter()) {
                case "1":
//                    final String[] items = new String[]{"蓝色（默认）", "黑色", "白色"};
//                    new QMUIDialog.MenuDialogBuilder(MainActivity.this)
//                            .addItems(items, new DialogInterface.OnClickListener() {
//                                @Override
//                                public void onClick(DialogInterface dialog, int which) {
//                                    log("点击了 " + which);
//                                    dialog.dismiss();
//                                }
//                            })
//                            .create()
//                            .show();
                    break;
                case "2":
                    new QMUIDialog.MessageDialogBuilder(MainActivity.this)
                            .setTitle("这个是标题")
                            .setMessage("确认下载此文件？")
                            .addAction("去掉", new QMUIDialogAction.ActionListener() {
                                @Override
                                public void onClick(QMUIDialog dialog, int index) {
                                    dialog.dismiss();
                                    log("点击了 " + index);
                                }
                            })
                            .addAction("确定", new QMUIDialogAction.ActionListener() {
                                @Override
                                public void onClick(QMUIDialog dialog, int index) {
                                    dialog.dismiss();
                                    log("点击了 " + index);
                                }
                            })
                            .show();
                    break;
            }
        });
    }

    public void log(String str) {
        Log.d(TAG, ">>>>>" + str);
        Toast.makeText(MainActivity.this, str, Toast.LENGTH_LONG).show();
    }
}

File: testdialog22/src/main/java/com/zzt/testdialog22/MainActivity.kt
package com.zzt.testdialog22

import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import androidx.recyclerview.widget.RecyclerView
import com.zzt.adapter.StartActivityRecyclerAdapter
import com.zzt.entity.StartActivityDao
import com.zzt.testdialog22.ww.DatePicker
import java.util.*

class MainActivity : AppCompatActivity() {
    var rv_data: RecyclerView? = null
    var diglogUtil: DialogUtil? = null
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        initView()
    }

    private fun initView() {
        rv_data = findViewById(R.id.rv_data)
        diglogUtil = DialogUtil()
        var daoList: MutableList<StartActivityDao> = mutableListOf()

        daoList.add(StartActivityDao("弹一个添加框", "", "1"))
        daoList.add(StartActivityDao("弹二个添加框", "", "2"))
        daoList.add(StartActivityDao("弹三个添加框", "", "3"))
        StartActivityRecyclerAdapter.setAdapterData(
            rv_data,
            RecyclerView.VERTICAL,
            daoList
        ) { itemView, position, data ->
            if (data is StartActivityDao) {
                when (data.arouter) {
                    "1" -> diglogUtil?.showDialog01(this@MainActivity)
                    "2" -> diglogUtil?.showDialog02(this@MainActivity)
                    "3" -> text3()
                }
            }
        }
    }

    private fun text3() {

        var datePicker = DatePicker(this@MainActivity)
        datePicker?.setLabel("", "", "")
        val date = Date()
        val year = date.year + 1900
        val month = date.month + 1
        datePicker?.setRangeStart(year - 100, month, date.date)
        datePicker?.setRangeEnd(year - 18, month, date.date)
        datePicker?.setSelectedItem(year - 18, month, date.date)

        datePicker?.show()
    }

}